samcli/lib/build/app_builder.py

```diff
@@ -342,7 +368,7 @@ def _build_layer(
         subfolder = get_layer_subfolder(specified_workflow)
 
         # artifacts directory will be created by the builder
-        artifacts_dir = str(pathlib.Path(self._build_dir, layer_name, subfolder))
+        artifacts_dir = str(pathlib.Path(artifacts_dir, subfolder))
```

Comment:
Why `layer_name` is removed from the path? Isn't this going to put built artifact directly in stack's build root folder?

Reply:
The artifacts_dir contains layer name now. I added a new argument `artifacts_dir` to `_build_layer()` to make it consistent with 

```
def _build_function(  # pylint: disable=R1710
        self,
        function_name: str,
        codeuri: str,
        packagetype: str,
        runtime: str,
        handler: Optional[str],
        artifacts_dir: str,
        metadata: Optional[Dict] = None,
    )
```

And `artifacts_dir` for both function and layer are created by app_builder, for example for `_build_layer()`:

<img width="1067" alt="image" src="https://user-images.githubusercontent.com/3804518/108121520-c553c800-7057-11eb-8368-a87e58539686.png">

Reply:
I updated the docstring of `_build_layer()` and hope it make it clearer

Reply:
Oh I see, but if that is the case, I would rather name the variable as `artifact_dir` (singular) to point that out.

Reply:
good idea, changed